src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```
@@ -0,0 +1,119 @@
+                        + " WHERE is_deleted = false AND id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateManufacturersStatement =
+                        connection.prepareStatement(updateStatement)) {
+            updateManufacturersStatement.setString(1, manufacturer.getName());
+            updateManufacturersStatement.setString(2, manufacturer.getCountry());
+            updateManufacturersStatement.setObject(3, manufacturer.getId());
+            updateManufacturersStatement.executeUpdate();
+            return manufacturer;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cant update manufacturer = "
+                    + manufacturer + " in DB", e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteStatement = "UPDATE manufacturers SET is_deleted = true WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteManufacturersStatement =
+                        connection.prepareStatement(deleteStatement)) {
+            deleteManufacturersStatement.setLong(1, id);
+            return deleteManufacturersStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cant delete manufacturer with id = "
+                    + id + " in DB", e);
+        }
+    }
+
+    private Manufacturer getManufacture(ResultSet resultSet) {
+        try {
+            Long id = resultSet.getObject("id", Long.class);
+            String name = resultSet.getString("name");
+            String country = resultSet.getString("country");
+            Manufacturer manufacturer = new Manufacturer(name, country);
+            manufacturer.setId(id);
+            return manufacturer;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cant get manufacturer from DB", e);
+        }
```

let's throw this exception on the method level